app/pages/ProjectsPage.tsx

```diff
@@ -64,9 +65,7 @@ export function Component() {
 
   const { mutateAsync: deleteProject } = useApiMutation('projectDelete', {
     onSuccess() {
-      // TODO: figure out if this is invalidating as expected, can we leave out the query
-      // altogether, etc. Look at whether limit param matters.
-      apiQueryClient.invalidateQueries('projectList')
+      invalidate('projectList')
```

Comment:
So, ideally we could do this as something like this:

```ts
queryClient.invalidateQueries(projectList.optionsFn())
```

or (to be clearer about what's going on)

```ts
queryClient.invalidateQueries({ queryKey: projectList.optionsFn().queryKey })
```

The problem here is that that `queryKey` has way too much specificity. Even without a page token specified, it will have the `limit` on it:

https://github.com/oxidecomputer/console/blob/e4f70b01b1c14fd194d306b45a1ca87efa0f1468/app/api/hooks.ts?plain=1#L168-L169

which means it won't invalidate any other fetches for the project list that have a different limit, and it won't invalidate any cached queries for pages other than the first one. This shows the wisdom of our approach of just using the method key alone without any parameters, which has the effect of invalidating any cached `projectList` fetch, regardless of the details.

I have an idea, though. If the list option object has another method that returns just the method key, we could use that.